tools/oneclickapps/README.md

```diff
@@ -0,0 +1,70 @@
+# One click app release automation
+
+This directory contains scripts for automating the release of Zulip one click apps.
+
+## DigitalOcean 1-Click Application
+`prepare_digital_ocean_one_click_app_release.py` creates the image of DigitalOcean 1-Click
+app from the latest Zulip release (fetched from https://www.zulip.org/dist/releases). It will
+also create a test droplet from the image and send the image and droplet
+details to a pre-configured Zulip stream. Anyone, whose key is added to the
+Zulip DigitalOcean team can SSH into the droplet for testing.
+
+### Running as GitHub action
+
+`.github/workflows/update-oneclick-apps.yml` is configured to invoke
+`prepare_digital_ocean_one_click_app_release.py` as a GitHub action during each Zulip
+server release.
+
+You also need to set the following secrets in your GitHub repository to make the action
+work correctly. These secrets are passed as environment variables to the GitHub action.
+
+* `DIGITALOCEAN_API_KEY` - DigitalOcean API KEY used for creating droplets, snapshots etc.
+* `ZULIP_SITE` - The URL of the Zulip realm where the message containing droplet/image details
+  should be sent.
+* `ZULIP_API_KEY` - The API key of the Zulip Bot used for sending messages.
+* `ZULIP_STREAM` - The stream to which the messages should be sent.
+* `ZULIP_EMAIL` - The email of the Zulip bot.
+
+Also pass the following as environment variables in `.github/workflows/update-oneclick-apps.yml`.
+*  `PYTHON_DIGITALOCEAN_REQUEST_TIMEOUT_SEC` -  This configures the maximum number of seconds
+  to wait before the requests made by `python-digitalocean` timeout. If not configured, it's
+  common for the requests to take 20+ minutes before getting timed out.
+
+### Verifying the one click app image
+* The action would send the image name and test droplet details to the stream configured in the
+  above steps.
+* SSH into the test droplet by following the instructions in the message.
+* After logging into the test droplet, exit the installer and run the following script.
+
+  https://raw.githubusercontent.com/digitalocean/marketplace-partners/master/scripts/img_check.sh
+
+  This script checks whether the image created is valid. It is also run by the DigitalOcean team
+  before they approve the image submission in the one click app marketplace.
+* If there are no errors (you can ignore most of the warnings), exit the SSH connection and
+  reconnect.
+* Populate the details asked by the installer and verify that the installer completes successfully.
+  If there are errors see the section below.
+* Use the link generated by the installer to create a new Zulip organization. Do some basic
+  testing like sending a bunch of messages and reloading the webpage.
```

Comment:
Can we also automate these steps as a follow-up?

Reply:
Yes. I think we could automate them using Fabric and a headless browser. I will create a follow-up issue to keep track once it is merged.